File: Elixir/rot13/lib/rot13.ex
defmodule Rot13 do
  def rotate({k, v}) do
    cond do
      v in ?A..?M ->
        k + 13

      v in ?N..?Z ->
        k - 13

      true ->
        k
    end
  end

  def main(_) do
    encoded =
      IO.gets("Enter string to encode:\n")
      |> String.trim()
      |> then(fn xs ->
        List.zip([
          xs
          |> String.to_charlist(),
          xs
          |> String.upcase()
          |> String.to_charlist()
        ])
      end)
      |> Enum.map(&rotate/1)

    IO.puts("Encoded string:\n#{encoded}")
  end
end

File: Erlang/rot13/src/rot13.erl
-module(rot13).

-export([main/1]).

rotate({K, V}) ->
    A = lists:member(V, lists:seq($A, $M)),
    B = lists:member(V, lists:seq($N, $Z)),
    if
        A ->
            K + 13;
        B ->
            K - 13;
        true ->
            K
    end.

main(_) ->
    Text = string:trim(io:get_line("Enter string to encode:\n")),
    List = lists:zip(Text, string:to_upper(Text)),
    Encoded = lists:map(fun rotate/1, List),
    io:format("Encoded string:~n~ts~n", [Encoded]).
